File: Example/Example/DateTransformer.m
//
//  ViewController.h
//  JSONObjectMapperExample
//
//  Created by Sergey on 20/05/2018.
//  Copyright © 2018 Sergey. All rights reserved.
//

#import "DateTransformer.h"

@interface DateTransformer ()

@property (nonatomic, readonly) NSDateFormatter *dateFormatter;

@end

@implementation DateTransformer

+ (Class)transformedValueClass {
    return [NSString class];
}

- (instancetype)initWithDateFormatter:(NSDateFormatter *)dateFormatter {
    if (self = [super init]) {
        _dateFormatter = dateFormatter;
    }
    return self;
}

+ (BOOL)allowsReverseTransformation {
    return YES;
}

- (id)transformedValue:(id)value {
    return [self.dateFormatter dateFromString:value];
}

- (id)reverseTransformedValue:(id)value {
    return [self.dateFormatter stringFromDate:value];
}

@end

File: ExampleSwift/ExampleSwift/DateTransformer.swift
//
//  DateTransformer.swift
//  JSONObjectMapper_Example
//
//  Created by Sergey on 21/08/2018.
//

import UIKit

/// DateTransformer converts strings to dates and vice versa.
///
/// You can configure which date format to use via the \c dateFormat parameter in the initializer.
///
class DateTransformer: ValueTransformer {
    
    let dateFormatter: DateFormatter
    
    override class func transformedValueClass() -> AnyClass {
        return NSString.self
    }
    
    override class func allowsReverseTransformation() -> Bool {
        return true
    }
    
    init(dateFormatter: DateFormatter) {
        self.dateFormatter = dateFormatter
    }
    
    override func transformedValue(_ value: Any?) -> Any? {
        return dateFormatter.date(from: (value as? String)!)
    }
    
    override func reverseTransformedValue(_ value: Any?) -> Any? {
        return dateFormatter.string(from: (value as? Date)!)
    }
}
